contracts/JoepegExchange.sol

```diff
@@ -211,6 +211,40 @@ contract JoepegExchange is
             msg.sender == takerBid.taker,
             "Order: Taker must be the sender"
         );
+    }
+
+    /**
+     * @notice Match ask with a taker bid order using AVAX and ignore expired asks if any
+     * @param takerBid taker bid order
+     * @param makerAsk maker ask order
+     */
+    function _matchAskWithTakerBidUsingAVAXAndWAVAXIgnoringExpiredAsks(
+        OrderTypes.TakerOrder calldata takerBid,
+        OrderTypes.MakerOrder calldata makerAsk
+    ) internal {
+        // Validate orders
+        _validateMakerAskAndTakerBid(takerBid, makerAsk);
```

Comment:
it seems like we call this method at the beginning of `_matchAskWithTakerBidUsingAVAXAndWAVAX` as well. could we just do it there instead of calling it twice?

Reply:
`_matchAskWithTakerBidUsingAVAXAndWAVAX` is also called from `matchAskWithTakerBidUsingAVAXAndWAVAX` and `batchBuyWithAVAXAndWAVAX`, we need to validate the orders in that case.

 We could move this call outside of `_matchAskWithTakerBidUsingAVAXAndWAVAX` and put it right before `_matchAsk...` calls in other methods but I thought it was safer to keep it in `_matchAsk...` (if we update the contract in the future and we forget that we need to call `_validateMakerAskAndTakerBid` first for example)